design/dhcp-less.md

```diff
@@ -0,0 +1,391 @@
+preprov-template hostSelector will have preprovisioningNetworkDataName assigned using the same process
+outlined above for `networkData` above.
+
+There are a few things to consider:
+
+To avoid the risk of multiple Metal3PreprovisioningTemplate resources matching BareMetalHost resources (which would be ambiguous)
+a BMH must match *exactly one* Metal3PreprovisioningTemplate for the conroller to take action, if more than one matches it will be
+reflected as ignored via the Metal3PreprovisioningTemplate status.
+
+The preprovisioningNetworkDataName is used by default for networkData in the baremetal-operator, so in this configuration it's not
+strictly necessary to specify networkData via Metal3DataTemplate, however we'll want to delete the IPClaim after preprovisioning 
+in the decoupled flow below so it seems likely we'll want to behave consistently and rely on the IP Reuse functionality if a
+consistent IP is required between pre-provisioning and provisioning phases.
+
+#### Preprovisioning Decoupled IPPool
+
+```yaml
+apiVersion: ipam.metal3.io/v1alpha1
+kind: IPPool
+metadata:
+  name: pool-1
+spec:
+  clusterName: cluster
+
+---
+
+apiVersion: ipam.metal3.io/v1alpha1
+kind: IPPool
+metadata:
+  name: preprovisioning-pool
+
+---
+
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3PreprovisioningDataTemplate
+metadata:
+  name: preprov-data-template
+spec:
+  preprovisioningNetworkData:
+    networks:
+      ipv4:
+      - id: eth0
+        ipAddressFromIPPool: preprovisioning-pool
+
+---
+
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3PreprovisioningTemplate
+metadata:
+  name: preprov-template
+spec:
+  template:
+    spec:
+      dataTemplate:
+        name: preprov-data-template
+      hostSelector:
+        matchLabels:
+          pre-provisioning: foo
+
+---
+
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3DataTemplate
+metadata:
+  name: data-template
+spec:
+  clusterName: cluster
+  networkData:
+    networks:
+      ipv4:
+      - id: eth0
+        ipAddressFromIPPool: pool-1
+
+---
+
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3MachineTemplate
+metadata:
+  name: machine-template
+spec:
+  template:
+    spec:
+      dataTemplate:
+        name: data-template
+      hostSelector:
+        matchLabels:
+          cluster-role: control-plane
+
+```
+
+In this flow we have `preprovisioning-pool` which is not associated with any cluster, this is used to provide an IPAddress during
+the pre-provisioning phase as described above.  To reduce the required size of the pool, the IPClaim will be deleted after the
+preprovisioning phase is completed, e.g the BMH resource becomes available.
+
+In the provisioning phase another pool, associated with a cluster is used to template networkData as in the existing process.
+
+#### Assumptions and Open Questions
+
+TODO
+
+### Inspection on initial registration
+
```

Comment:
Perhaps the simplest solution would be to add e.g `PreprovisioningNetworkDataRequired` to the BMH API - this provides a clear way to describe that the host cannot move from Registering -> Inspecting and avoids coupling the BMH to a template CR (it's valid to use Secrets instead of templates, so this would handle both cases and keep the templating external to the BMH API surface)

@dtantsur @zaneb interested in your thoughts on this when you get a moment 😄

Reply:
I added `PreprovisioningNetworkDataRequired` as the proposed solution, but for now left the paused annotation described as an alternative

Reply:
Proposal adjusted to use a new BMH annotation as discussed elsewhere